### PR TITLE
Add dev.geopjr.Hashbrown

### PR DESCRIPTION
**Pull request by GeopJr**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by GeopJr** [comment]
bot, build dev.geopjr.Hashbrown

**Comment by GeopJr** [comment]
bot, build dev.geopjr.Hashbrown

**Comment by GeopJr** [comment]
bot, build dev.geopjr.Hashbrown

File: dev.geopjr.Hashbrown.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+runtime: org.gnome.Platform
+runtime-version: "3.38"
+sdk: org.gnome.Sdk
+command: hashbrown
+finish-args:
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --share=ipc
+  - --device=dri
+  - --filesystem=host:ro
```
Comment (barthalion) [inline dev.geopjr.Hashbrown.yaml:11]: You should be able to use `GtkFileChooserNative` which transparently uses portal for accessing single files.

Comment (GeopJr) [reply dev.geopjr.Hashbrown.yaml]: Thanks!

**Comment by GeopJr** [comment]
bot, build dev.geopjr.Hashbrown

File: dev.geopjr.Hashbrown.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        url: https://github.com/GeopJr/Hashbrown.git
+        tag: v1.0.2
+        commit: 09c2998521ed5b14560b12ebb503bc4835dd9009
+      - type: archive
+        dest: crystal/
+        url: https://github.com/crystal-lang/crystal/releases/download/0.35.1/crystal-0.35.1-1-linux-x86_64.tar.gz
+        sha256: 6c3fd36073b32907301b0a9aeafd7c8d3e9b9ba6e424ae91ba0c5106dc23f7f9
+      - type: git
+        url: https://github.com/jhass/crystal-gobject.git
+        branch: main
```
Comment (nedrichards) [inline dev.geopjr.Hashbrown.yaml:47]: this needs to point to a specific git commit/tag for build reproducibility as per https://github.com/flathub/flathub/wiki/App-Requirements#stable-releases-reproducible-builds

File: dev.geopjr.Hashbrown.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        dest: lib/gobject
+      - type: git
+        url: https://github.com/elorest/compiled_license.git
+        branch: master
```
Comment (nedrichards) [inline dev.geopjr.Hashbrown.yaml:51]: this needs to point to a specific git commit/tag for build reproducibility as per https://github.com/flathub/flathub/wiki/App-Requirements#stable-releases-reproducible-builds

**Comment by nedrichards** [comment]
Other than the points above, lgtm

File: dev.geopjr.Hashbrown.yaml (hunk 2 above)
Comment (GeopJr) [reply dev.geopjr.Hashbrown.yaml]: The latest tag is missing some important commits, so I will only specify the latest commit.

File: dev.geopjr.Hashbrown.yaml (hunk 3 above)
Comment (GeopJr) [reply dev.geopjr.Hashbrown.yaml]: Done, thanks!

**Comment by GeopJr** [comment]
bot, build dev.geopjr.Hashbrown

**Pull request closed without merging**